main.go

```diff
@@ -17,6 +23,95 @@ import (
 	"github.com/urfave/cli/v2"
 )
 
+type MigrationRunner struct {
+	path string
+}
+
+func NewMigrationRunner(path string) *MigrationRunner {
+	return &MigrationRunner{path: path}
+}
+
+func (m *MigrationRunner) Run(ctx context.Context, conn *sql.Conn, lastRanId int) error {
+	files, err := m.loadMigrationFiles()
+	if err != nil {
+		return err
+	}
+
+	sort.Slice(files, func(i, j int) bool {
+		id1 := extractMigrationID(files[i])
+		id2 := extractMigrationID(files[j])
+
+		return id1 < id2
+	})
+
+	// var index int = 0
+	// for _, file := range files {
+	// 	if extractMigrationID(file) <= lastRanId {
+	// 		index++
+	// 	}
+	// }
+
+	fileSlice := files[lastRanId+1:]
+
+	for _, file := range fileSlice {
+		readFile, err := os.Open(file)
+
+		if err != nil {
+			fmt.Println(err)
+		}
+		fileScanner := bufio.NewScanner(readFile)
+
+		fileScanner.Split(bufio.ScanLines)
+		for fileScanner.Scan() {
+
+			if strings.Trim(fileScanner.Text(), "") == "" {
+				continue
+			}
+			if _, err := conn.ExecContext(ctx, fileScanner.Text()); err != nil {
+				return fmt.Errorf("failed to execute migration %s: %w", file, err)
+			}
+		}
+
+		readFile.Close()
+
+		log.Printf("Executed migration: %s\n", file)
+	}
+
+	return nil
+}
+
+func extractMigrationID(file string) int {
+	regex := regexp.MustCompile(`_(\d+)\.sql$`)
+	matches := regex.FindStringSubmatch(file)
+	if len(matches) != 2 {
+		return -1
+	}
+
+	id, err := strconv.Atoi(matches[1])
+	if err != nil {
+		return -1
+	}
+
+	return id
+}
```

Comment:
A good rule of thumb for go is that if it's possible for your function to fail, return an error. Even though what you're doing here with returning a -1 is working (and it's clever), there's a really common lesson each developer needs to learn: [clear is better than clever](https://dave.cheney.net/2019/07/09/clear-is-better-than-clever). 

I'd suggest this function return `(int, error)`, and our migration runner just errors out if there's a bad file name in the `migrations` directory.